ui/src/components/SampleView/ContextMenu.jsx

```diff
@@ -160,28 +159,36 @@ export default function ContextMenu(props) {
       }
     });
 
-    const options = {
+    return {
       SAVED: [
-        {
-          text: 'Add Datacollection',
-          action: () => showModal('DataCollection'),
-          key: 'datacollection',
-        },
-        {
-          text: 'Add Characterisation',
-          action: () => showModal('Characterisation'),
-          key: 'characterisation',
-        },
-        {
-          text: 'Add XRF Scan',
-          action: () => showModal('xrf_spectrum'),
-          key: 'xrf_spectrum',
-        },
-        {
-          text: 'Add Energy Scan',
-          action: () => showModal('energy_scan'),
-          key: 'energy_scan',
-        },
+        availableMethods.has('datacollection')
```

Comment:
I purposely did not to keep a filtering loop (cf. below) as the goal is to make this component more declarative.